packages/flutter/lib/src/widgets/scroll_physics.dart

```diff
@@ -529,15 +533,23 @@ class RangeMaintainingScrollPhysics extends ScrollPhysics {
       enforceBoundary = false;
     }
     if (maintainOverscroll) {
-      // Force the new position to be no more out of range
-      // than it was before, if it was overscrolled.
-      if (oldPosition.pixels < oldPosition.minScrollExtent) {
-        final double oldDelta = oldPosition.minScrollExtent - oldPosition.pixels;
-        return newPosition.minScrollExtent - oldDelta;
-      }
-      if (oldPosition.pixels > oldPosition.maxScrollExtent) {
-        final double oldDelta = oldPosition.pixels - oldPosition.maxScrollExtent;
-        return newPosition.maxScrollExtent + oldDelta;
+      // Force the new position to be no more out of range than it was before, if:
+      //  * it was overscrolled
+      //  * and the position has shrunk, meaning that some content was removed. The
```

Comment:
maybe rather than "the position has shrunk" we should say "the position has decreased".

That said, I think you mean the extents have shrunk? rather than the position. The position is what we're changing or not changing, not what we're basing it on. (It's a bit confusing because the extents are on a variable called "position", I'll grant you.)

Reply:
Done. Changed to "the extents have decreased"